packages/babel-helpers/src/helpers.js

```diff
@@ -78,9 +78,10 @@ helpers.jsx = helper("7.0.0-beta.0")`
 `;
 
 helpers.asyncIterator = helper("7.0.0-beta.0")`
+  const symbolTypes = ["function", "object"];
   export default function _asyncIterator(iterable) {
     var method
-    if (typeof Symbol === "function") {
+    if (symbolTypes.includes(typeof Symbol)) {
```

Comment:
Maybe this could just be `typeof Symbol !== "undefined"`?

Reply:
Sounds good to me